lib/repository.js

```diff
@@ -1094,6 +1095,23 @@ Repository.prototype.getStatusExt = function(opts) {
 };
 
 /**
+ * Get the names of the submodules in the repository.
+ *
+ * @async
+ * @return {Array<String>}
+ */
+Repository.prototype.getSubmoduleNames = function() {
+  var names = [];
+  var callback = function(submodule, name, payload) {
+    names.push(name);
+  };
+
+  return Submodule.foreach(this, callback).then(function() {
+    return names;
```

Comment:
We also support using callbacks as well. Reference https://github.com/nodegit/nodegit/blob/master/lib/repository.js#L98-L100 for an example on implementing.